run_xnat2bids.py

```diff
@@ -392,12 +438,60 @@ async def main():
         })
         
         # Run xnat2bids asynchronously.
-        task = asyncio.create_task(asyncio.create_subprocess_exec(*sbatch_cmd))
+        task = asyncio.create_task(asyncio.create_subprocess_exec(*sbatch_cmd, stdout=f))
```

Comment:
Redirect only!

Reply:
Ok! It actually might be helpful to print out the job IDs, since we have them. People can always get them with myq, but it's nice to have them in case they want to monitor or cancel a certain job. Anyone who uses sbatch is used to it echoing the launched job ID when they submit.

Reply:
Great idea! I am logging the Job-IDs as follows: 

```
INFO: Launched 4 jobs
INFO: Job IDs: 9941173 9941174 9941175 9941176
```

Reply:
It looks like it'll print the job IDs for the bids validator jobs, but not the xnat2bids jobs; is that correct?

Reply:
I'm using a little Python trick, in that default parameters are mutable.  From Codecademy, "Python's default arguments are evaluated only once when the function is defined, not each time the function is called. This means that if a mutable default argument is used and is mutated, it is mutated for all future calls to the function as well."

So the default parameter "jobs" is updated and picked up where it left off on the second call to fetch_job_ids().  This should usually be avoided, as it can cause unexpected behavior.  In this case, I'm leveraging the quirky aspect of Python to update jobs after launching the bids-validator.  I'll try to brainstorm a more intuitive, maintainable solution.

Reply:
Technically, I wouldn't even have to set `jobs = fetch_job_ids(validator_output)`, but only call `fetch_job_ids(validator_output)`.  I just wanted it to be clear that `jobs` is being updated.

Reply:
I removed the default parameter, initializing a new jobs list with every call to fetch_job_ids(), and I am using a function called update_jobs(jobs, output) to extend the existing jobs list.  I think this makes more sense. 

```
def fetch_job_ids(stdout):
    jobs = []
    if isinstance(stdout, bytes):
        stdout = [stdout]

    for line in stdout:
        job_id = line.replace(b'Submitted batch job ', b'' ).strip().decode('utf-8')
        jobs.append(job_id)

    return jobs

def update_jobs(jobs, output):
    new_jobs = fetch_job_ids(output)
    jobs.extend(new_jobs)
```

Reply:
Oh gotcha, so the logging would print all the xnat2bids job IDs and then all the validator job IDs? Could we just do something like `valjobs = fetch_job_ids(validator_output)`

and then an additional logging statement like
```
logging.info("After these jobs complete, BIDS compliance will be validated:")
logging.info("Job %s: %s", "IDs" if len(valjobs) > 1 else "ID", ' '.join(valjobs))
```